apps/server/src/modules/account/repo/account.repo.integration.spec.ts

```
@@ -44,183 +33,340 @@ describe('account repo', () => {
+				const accounts = await repo.findMultipleByUserId([anAccountToFind.userId!, anotherAccountToFind.userId!]);
+				expect(accounts).toContainEqual(anAccountToFind);
+				expect(accounts).toContainEqual(anotherAccountToFind);
+				expect(accounts).toHaveLength(2);
+			});
 		});
 
-		it('should return empty list if no results', async () => {
-			const accountToFind = accountFactory.build();
-			await em.persistAndFlush(accountToFind);
-			em.clear();
-			const accounts = await repo.findMultipleByUserId(['123456789012', '098765432101']);
-			expect(accounts).toHaveLength(0);
+		describe('When not existing user ids are given', () => {
+			it('should return empty list', async () => {
+				const accounts = await repo.findMultipleByUserId(['123456789012', '098765432101']);
+				expect(accounts).toHaveLength(0);
+			});
 		});
 	});
 
 	describe('findByUserIdOrFail', () => {
-		it('should find a user by id', async () => {
-			const accountToFind = accountFactory.build();
-			await em.persistAndFlush(accountToFind);
-			em.clear();
-			const account = await repo.findByUserIdOrFail(accountToFind.userId ?? '');
-			expect(account.id).toEqual(accountToFind.id);
+		describe('When existing id is given', () => {
+			const setup = async () => {
+				const accountToFind = accountFactory.build();
+				await em.persistAndFlush(accountToFind);
+				em.clear();
+				return accountToFind;
+			};
+			it('should find a user', async () => {
+				const accountToFind = await setup();
+				const account = await repo.findByUserIdOrFail(accountToFind.userId ?? '');
+				expect(account.id).toEqual(accountToFind.id);
+			});
 		});
 
-		it('should throw if id does not exist', async () => {
-			const accountToFind = accountFactory.build();
-			await em.persistAndFlush(accountToFind);
-			em.clear();
-			await expect(repo.findByUserIdOrFail('123456789012')).rejects.toThrow(NotFoundError);
+		describe('When id does not exist', () => {
+			it('should throw', async () => {
```

should throw not found error